sql/catalyst/src/main/scala/org/apache/spark/sql/connector/catalog/CatalogManager.scala

```diff
@@ -44,15 +44,27 @@ class CatalogManager(
   import CatalogManager.SESSION_CATALOG_NAME
 
   private val catalogs = mutable.HashMap.empty[String, CatalogPlugin]
+  // Map from catalog back to it's original name for easy name look up, we don't use the
+  // CatalogPlugin's name as it might be different from the catalog name depending on
+  // implementation.
+  private val catalogIdentifiers = mutable.HashMap.empty[CatalogPlugin, String]
```

Comment:
Do we need to handle `v1SessionCatalog` passed by the constructor here too?

Reply:
no, that catalog doesn't have a name

Reply:
I'd add a comment here stating that we will have a new instance of a catalog for each catalog name, therefore this reverse map works as intended. We should also add a test for it

Reply:
Why is the reverse map needed? Can't we just call `CatalogPlugin.name`?

Reply:
According to my understand `CatalogPlugin.name` depends on the underlying implementation which might not be the actual identifier name for the catalog.

Reply:
The plugin name is intended to prevent needing to do this. While we do rely on the catalog not to report the wrong name, I think it is reasonable to use it. I'm not strongly against this, though. If you think this is cleaner we can do that.